confmap/confmap.go

```diff
@@ -244,6 +259,21 @@ func castTo(exp expandedValue, useOriginal bool) (any, error) {
 	return exp.Value, nil
 }
 
+// Check if a reflect.Type is of the form T:
+// T = string | map[string]T | []T | [n]T
+func isStringyStructure(t reflect.Type) bool {
+	if t.Kind() == reflect.String {
+		return true
+	}
+	if t.Kind() == reflect.Map {
+		return t.Key().Kind() == reflect.String && isStringyStructure(t.Elem())
```

Comment:
Why do we need to require keys to be strings?

Reply:
It's not really needed, dropped that requirement